### PR TITLE
Report process init time for app start

### PR DESCRIPTION
**Pull request by markushi**
## :scroll: Description

Example event: https://sentry-sdks.sentry.io/performance/sentry-android:5173ff17f6644be5b5fe016352349679
op: `process.load`, description: `"Process Initialization"`

image

## :bulb: Motivation and Context
To better understand the app start, we want to surface the "system" potion of app start as well. It's mainly unactionable, but showcases how long it takes the system to start the app-process, load some resources and finally execute app specific code. 

Fixes: https://github.com/getsentry/sentry-java/issues/3152

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by markushi** [comment]
@narsaynorath that's how this system span on Android would look like. Please let me know if the op/description would be fine for you.

**Comment by narsaynorath** [comment]
Perfect, the system span looks good to me!

afaik in the last design conversation we had around the app start UI, we're going to drop the system v. application widget, but this would likely be a span that we would surface in the span op table for users to see the change there

File: sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java [hunk 1]
```diff
@@ -155,6 +156,23 @@ private void attachColdAppStartSpans(
...
+    // Process init
+    final long classInitUptimeMs = appStartMetrics.getClassLoadedUptimeMs();
+    final @NotNull TimeSpan appStartTimeSpan = appStartMetrics.getAppStartTimeSpan();
+    if (appStartTimeSpan.hasStarted()) {
+      final @NotNull TimeSpan processInitTimeSpan = new TimeSpan();
+      processInitTimeSpan.setStartedAt(appStartTimeSpan.getStartUptimeMs());
+      processInitTimeSpan.setStartUnixTimeMs(appStartTimeSpan.getStartTimestampMs());
+
+      processInitTimeSpan.setStoppedAt(classInitUptimeMs);
+      processInitTimeSpan.setDescription("Process Initialization");
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:168]: Should we add a guard in cases the app process is created a lot of time before the app is started (e.g. a notification or a service)?

Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:168]: That's a good point! iOS already discards app starts longer than a certain period - let me have a look into this!

Comment (stefanosiano) [reply sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:168]: @markushi did you have the chance to look into this?

Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:171]: 👍 I've added an extra check here: https://github.com/getsentry/sentry-java/pull/3159/commits/4e16c88756164c7d71fce7cb07ec514561be7ed0

File: sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java [hunk 2]
```diff
@@ -155,6 +157,25 @@ private void attachColdAppStartSpans(
       }
     }
 
+    // Process init
+    final long classInitUptimeMs = appStartMetrics.getClassLoadedUptimeMs();
+    final @NotNull TimeSpan appStartTimeSpan = appStartMetrics.getAppStartTimeSpan();
+    if (appStartTimeSpan.hasStarted()
+        && Math.abs(classInitUptimeMs - appStartTimeSpan.getStartUptimeMs())
+            <= MAX_PROCESS_INIT_APP_START_DIFF_MS) {
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:165]: looks good now, just add a comment for your future self 😄

**Review by stefanosiano — APPROVED** [review]

**Pull request merged**